storm/src/main/storm/mesos/MesosSupervisor.java

```diff
@@ -56,17 +56,30 @@
   Map _conf;
   // Store state on port assignments arriving from MesosNimbus as task-launching requests.
   private static final TaskAssignments _taskAssignments = TaskAssignments.getInstance();
-  // What is the storm-core supervisor's view of the assigned ports?
-  AtomicReference<Set<Integer>> _supervisorViewOfAssignedPorts = new AtomicReference<Set<Integer>>(new HashSet<Integer>());
 
   public static void main(String[] args) {
-    org.apache.storm.daemon.supervisor.launch(new MesosSupervisor());
+    Map<String, Object> conf = ConfigUtils.readStormConfig();
+
+    try {
+      Supervisor supervisor = new Supervisor(conf, null, new MesosSupervisor());
+
+      // We use reflection to call the private method, 'launchDaemon', which calls 'launch'
+      Method m = Supervisor.class.getDeclaredMethod("launchDaemon");
```

Comment:
Whoa... I missed that `launchDaemon` was *private*!!  I'll propose changing its visibility when I resurrect `ISupervisor.assigned()` in storm-core proper, as you can track here:
* https://issues.apache.org/jira/browse/STORM-2690